### PR TITLE
feat(GA): add new resource ga_accelerator

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

add new resource ga_accelerator

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
add new resource ga_accelerator
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/ga' TESTARGS='-run TestAccAccelerator_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ga -v -run TestAccAccelerator_basic -timeout 360m -parallel 4
=== RUN   TestAccAccelerator_basic
=== PAUSE TestAccAccelerator_basic
=== CONT  TestAccAccelerator_basic
--- PASS: TestAccAccelerator_basic (52.78s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ga        52.855s
```

### CODE REVIEW

File: docs/resources/ga_accelerator.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+* `description` - (Optional, String) Specifies the description about the global accelerator. The value can contain
+  0 to 255 characters. The following characters are not allowed: <>
+
+* `enterprise_project_id` - (Optional, String, ForceNew) Specifies the enterprise project ID of the tenant.
+  The value is **0** or a string that contains a maximum of 36 characters in UUID format with hyphens (-).
+  **0** indicates the default enterprise project. Defaults to **0**.
+
+  Changing this parameter will create a new resource.
+
+* `tags` - (Optional, List, ForceNew) Specifies the tags.
```
Comment (Lance52259) [inline docs/resources/ga_accelerator.md]:
```suggestion
* `tags` - (Optional, Map, ForceNew) Specifies the key/value pairs to associate with the global accelerator.
```

File: docs/resources/ga_accelerator.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  + **1**: frozen (Resources and data will be retained, but the cloud service cannot be used.)
+  + **2**: deleted/terminated (Both resources and data will be cleared.)
+
+## Timeouts
+
+This resource provides the following timeouts configuration options:
+
+* `create` - Default is 10 minute.
+* `update` - Default is 10 minute.
+* `delete` - Default is 10 minute.
```
Comment (Lance52259) [inline docs/resources/ga_accelerator.md]:
```suggestion
* `create` - Default is 10 minutes.
* `update` - Default is 10 minutes.
* `delete` - Default is 10 minutes.
```

File: docs/resources/ga_accelerator.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+## Import
+
+The accelerator can be imported using the `id`, e.g.
+
+```
+$ terraform import huaweicloud_ga_accelerator.test 0ce123456a00f2591fabc00385ff1234
```
Comment (Lance52259) [inline docs/resources/ga_accelerator.md]: actually, the ID format is `xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`.

File: huaweicloud/utils/tags.go [hunk 4]
```diff
@@ -70,12 +70,12 @@ func TagsToMap(tags []tags.ResourceTag) map[string]string {
...
-func FlattenTagsToMap(tags interface{}) map[string]string {
+func FlattenTagsToMap(tags interface{}) map[string]interface{} {
 	if tagArray, ok := tags.([]interface{}); ok {
-		result := make(map[string]string)
+		result := make(map[string]interface{})
 		for _, val := range tagArray {
-			if t, ok := val.(map[string]string); ok {
-				result[t["key"]] = t["value"]
+			if t, ok := val.(map[string]interface{}); ok {
+				result[t["key"].(string)] = t["value"]
```
Comment (Lance52259) [inline huaweicloud/utils/tags.go:78]: Perhaps it would be better in another commit.

File: huaweicloud/utils/tags.go [hunk 5]
```diff
@@ -99,6 +99,21 @@ func ExpandResourceTags(tagmap map[string]interface{}) []tags.ResourceTag {
...
 }
 
+// ExpandResourceTagsMap returns the tags in format of list of maps for the given map of data.
+func ExpandResourceTagsMap(tagmap map[string]interface{}) []map[string]interface{} {
+	var taglist []map[string]interface{}
+
+	for k, v := range tagmap {
+		tag := map[string]interface{}{
+			"key":   k,
+			"value": v.(string),
```
Comment (Lance52259) [inline huaweicloud/utils/tags.go]:
```suggestion
			"value": v,
```

File: huaweicloud/utils/tags.go [hunk 6]
```diff
@@ -99,6 +99,21 @@ func ExpandResourceTags(tagmap map[string]interface{}) []tags.ResourceTag {
 	return taglist
 }
 
+// ExpandResourceTagsMap returns the tags in format of list of maps for the given map of data.
+func ExpandResourceTagsMap(tagmap map[string]interface{}) []map[string]interface{} {
+	var taglist []map[string]interface{}
```
Comment (Lance52259) [inline huaweicloud/utils/tags.go]:
```suggestion
	if len(tagmap) < 1 {
		return nil
	}
	taglist := make([]map[string]interface{}, 0, len(tagmap))
```

File: huaweicloud/services/ga/resource_huaweicloud_ga_accelerator.go [hunk 7]
```diff
@@ -0,0 +1,694 @@
...
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the global accelerator name. The name can contain 1 to 64 characters. Only letters, digits, and hyphens (-) are allowed.`,
```
Comment (Lance52259) [inline huaweicloud/services/ga/resource_huaweicloud_ga_accelerator.go]: too long

File: huaweicloud/services/ga/resource_huaweicloud_ga_accelerator.go [hunk 8]
```diff
@@ -0,0 +1,694 @@
...
+			"flavor_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Specifies the specification ID.`,
+			},
+			"frozen_info": {
+				Type:     schema.TypeList,
+				Elem:     AcceleratorFrozenInfoSchema(),
+				Computed: true,
+			},
```
Comment (Lance52259) [inline huaweicloud/services/ga/resource_huaweicloud_ga_accelerator.go]: Seems the description is missing.

File: huaweicloud/services/ga/resource_huaweicloud_ga_accelerator.go [hunk 9]
```diff
@@ -0,0 +1,694 @@
...
+			"tags": common.TagsForceNewSchema(),
+
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Description: `Specifies the provisioning status. The value can be one of the following:
+- **ACTIVE**: The resource is running.
+- **PENDING**: The status is to be determined.
+- **ERROR**: Failed to create the resource.
+- **DELETING**: The resource is being deleted.`,
```
Comment (Lance52259) [inline huaweicloud/services/ga/resource_huaweicloud_ga_accelerator.go]: Perhaps it would be better to mark the end with two spaces in the .yaml file.
```
Specifies the provisioning status.  
The value can be one of the following:
...
```

File: docs/resources/ga_accelerator.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/resources/ga_accelerator.md]: fixed

File: docs/resources/ga_accelerator.md (hunk 2 above)
Comment (Jason-Zhang9309) [reply docs/resources/ga_accelerator.md]: fixed

File: docs/resources/ga_accelerator.md (hunk 3 above)
Comment (Jason-Zhang9309) [reply docs/resources/ga_accelerator.md]: fixed

File: huaweicloud/utils/tags.go (hunk 4 above)
Comment (Jason-Zhang9309) [reply huaweicloud/utils/tags.go:78]: fixed

File: huaweicloud/utils/tags.go (hunk 5 above)
Comment (Jason-Zhang9309) [reply huaweicloud/utils/tags.go]: fixed

File: huaweicloud/utils/tags.go (hunk 6 above)
Comment (Jason-Zhang9309) [reply huaweicloud/utils/tags.go]: fixed

File: huaweicloud/services/ga/resource_huaweicloud_ga_accelerator.go (hunk 7 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/ga/resource_huaweicloud_ga_accelerator.go]: fixed

File: huaweicloud/services/ga/resource_huaweicloud_ga_accelerator.go (hunk 8 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/ga/resource_huaweicloud_ga_accelerator.go]: fixed

File: huaweicloud/services/ga/resource_huaweicloud_ga_accelerator.go (hunk 9 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/ga/resource_huaweicloud_ga_accelerator.go]: fixed

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/ga' TESTARGS='-run TestAccAccelerator_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ga -v -run TestAccAccelerator_basic -timeout 360m -parallel 4
=== RUN   TestAccAccelerator_basic
=== PAUSE TestAccAccelerator_basic
=== CONT  TestAccAccelerator_basic
--- PASS: TestAccAccelerator_basic (61.37s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ga        61.488s
```
